test/replication.test.js

```diff
@@ -52,6 +52,121 @@ describe('Replication / Change APIs', function() {
     };
   });
 
+  describe('optimization check rectifyChange Vs rectifyAllChanges', function() {
+    beforeEach(function initialData(done) {
+      var data = [{name: 'foo1', lname: 'bar1'}, {name: 'foo2', lname: 'bar2'}];
+      async.auto({
+        createSourceModel: function(callback) {
+          SourceModel.create(data, callback);
+        },
+        replicate: ['createSourceModel', function(callback, results) {
+            SourceModel.replicate(TargetModel, callback);
+          }]
+      }, function(err, results) {
+        done(err);
+      });
+    });
+
+    it('rectifyOnDelete for Delete should call rectifyChange instead of rectifyAllChanges', function(done) {
+      var count = 0;
+      TargetModel.rectifyChange = function() {
+        // based on destroyAll where condition it should call rectifyChange two times
+        if (++count == 2) {
+          TargetModel.find(function(err, data) {
+            expect(data.length).to.equal(0);
+            return done(err);
+          });
+        }
+      };
+      TargetModel.rectifyAllChanges = function() {
+        return done(new Error('Should not call rectifyAllChanges'));
+      };
+
+      async.auto({
+        destroySourceModel: function(callback) {
+          SourceModel.destroyAll({'or': [{name: 'foo1'}, {lname: 'bar2'}]}, callback);
```

Comment:
Hi Mirolsav,

> Always use the simplest solution that can reproduce the scenario you want to test.
Sure I make it simpler...

> It is good idea to keep these two high-level tests in place, please add also another one for CREATE, to get a complete coverage.
I already added the test case for CREATE -> https://github.com/strongloop/loopback/pull/1899/files#diff-1884851c88410a9acc23f7bb4dc6c6cfR132

>```
var calls = [];
SourceModel.rectifyChange = function() { calls.push('rectifyChange'); };
SourceModel.rectifyAllChanges = function() { calls.push('rectifyAllChanges'); };

await SourceModel.destroyAll({'or': [{name: 'foo1'}, {lname: 'bar2'}]}

expect(calls).to.eql(['rectifyAllChanges']); // no id was specified in the query
```
Sorry I did not get what you mean by `await` here... Is this a sudo code block or `await` is something I'm not familiar with...?